### PR TITLE
DYN-3777: disable linter choice in forbidden workspaces

### PR DESCRIPTION
**Pull request by BogdanZavu**
### Purpose
For now, we will only allow users to change Linter in HomeWorkspaceModels. This is temporary and we will remove this limitation in the near future.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### CODE REVIEW

File: src/LintingViewExtension/LinterViewModel.cs [hunk 1]
```diff
@@ -284,6 +291,11 @@ public void Dispose()
         {
             this.linterManager.RuleEvaluationResults.CollectionChanged -= RuleEvaluationResultsCollectionChanged;
             this.linterManager.PropertyChanged -= OnLinterManagerPropertyChange;
+            viewLoadedParams.CurrentWorkspaceChanged -= OnCurrentWorkspaceChanged;
+        }
+
+        public bool CanChangeLinter {
```
Comment (mjkkirschner) [inline src/LintingViewExtension/LinterViewModel.cs:297]: summary on public property please.

**Comment by mjkkirschner** [comment]
Can a user get into a situation where they open a custom node which is somehow set to the wrong linter and it cannot be disabled?

**Comment by saintentropy** [comment]
LGTM @BogdanZavu

**Review by saintentropy — APPROVED** [review]

**Comment by BogdanZavu** [comment]
The test failure is unrelated to this change and it passed when the build was triggered manually from Jenkins.

**Pull request merged**